.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+name: CI
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ["3.3", "3.2", "3.1", "3.0", "2.7"]
```

Comment:
In our [reusable gem-test flow](https://github.com/theforeman/actions?tab=readme-ov-file#gem-test) we use [ekohl/ruby-version](https://github.com/ekohl/ruby-version) to automatically set up the test matrix based on `required_ruby_version` in the gemspec. If you drop the `rest_client_version` complexity you can use that same flow for reduced maintenance.